### ISSUE TITLE
[Feature Request] Improve `Update failed` error message

### ISSUE THREAD

**Issue by ShadyAbuKalam** [ScoopInstaller/Scoop#4675]
#### Current Behavior

Issuing scoop update, gives the following
Updating Scoop...
Update failed.
#### Expected Behavior

scoop itself is updated
### System details

**Windows version:** [e.g. 7, 8, 10]
10
**OS architecture:** [e.g. 32bit, 64bit]
64bit
**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]
5.1.18362.1801
**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

Empty file

**Comment by ShadyAbuKalam** [comment]
I checked the configuration file from CMD and it's empty, but from powershell I have the following:
```
{
    "SCOOP_BRANCH":  "master",
    "SCOOP_REPO":  "https://github.com/ScoopInstaller/Scoop",
    "lastupdate":  "2022-01-12T18:15:24.5024135+02:00"
}
```
I tried to git clone other repos directly and it failed silently also. I think my git installation has an issue. 
I changed the current junction of git to an older version "2.33.1" and it worked, then updated git to "2.34.1" and it worked again.

I think, there was something wrong with git, but shall not scoop indicate that there is an issue ?

**Comment by rashil2000** [comment]
> but shall not scoop indicate that there is an issue ?

Hmm, yes, the error message can be improved.

**Comment by rashil2000** [comment]
@niheaven I think this still needs work, not fixed yet

**Comment by darthwalsh** [comment]
I've had the problem with "You have unstaged changes." on one machine, and now on another machine. The workaround to remove and add two buckets worked, but it feels like ScoopInstaller/Main#3692 might not be fixed? scoop v0.3.1. 

The cause might be a script I run every night, that basically runs `scoop update * ; scoop cleanup --cache *` and then reboots if there's a pending reboot?

**Comment by rashil2000** [comment]
It should only happen once; we pushed a fix for it. Is it happening again and again on the same device?

**Comment by darthwalsh** [comment]
@rashil2000 Yes, it happened again and again until I manually ran the bucket rm bucket add workaround.

The "You have unstaged changes." problem happened the first time on 2022-03-09, and happened every night the script ran (218 times) until I manually applied the workaround. (I haven't used this laptop much last year.)

The error was with the main and extras bucket, so I'm not 100% confident I was running an updated version of scoop all this time. But I see in my command history I ran `scoop --version` and got "v0.3.1" before I did the workaround to fix the broken buckets.

**Comment by rashil2000** [comment]
No, I was asking whether it happens again _after_ you apply the rm bucket workaround. The rm bucket workaround was needed (just once) when we pushed a fix for this problem.
